src/extension/providers/webln/index.ts

```diff
@@ -109,6 +113,27 @@ export default class WebLNProvider {
     });
   }
 
+  //webln events
+  on(...args: Parameters<EventEmitter["on"]>) {
```

Comment:
All this new code is duplicated in the two providers. I wonder if we can extend a common provider class or something. We could also move the common stuff like enabling and executing there.

I would do this as a follow up PR I guess.